autotest/cpp/test_gdal_jpegxr.cpp

```diff
@@ -0,0 +1,389 @@
+///////////////////////////////////////////////////////////////////////////////
```

Comment:
I'd try to limit the number of tests written in C++ to what cannot be tested through the Python API. In your checklist you mention adding Python tests for the driver. Either C++ or Python, but both look overkill.